x-pack/solutions/observability/plugins/streams/server/lib/streams/errors/simulation_failed_error.ts

```diff
@@ -6,15 +6,17 @@
  */
 
 import { errors } from '@elastic/elasticsearch';
+import { StatusError } from './status_error';
 
-export class SimulationFailed extends Error {
+export class SimulationFailedError extends StatusError {
   constructor(error: errors.ResponseError) {
     super(
       error.body?.error?.reason ||
         error.body?.error?.caused_by?.reason ||
         error.message ||
-        'Unknown error'
+        'Unknown error',
+      400
```

Comment:
Seems like this is not necessarily a 400, should we just forward what's coming from Elasticsearch here instead?

Reply:
I think this is how they were interpreted before, but happy to change, i think that makes more sense

Reply:
I've changed this to propagate the ES error

Reply:
This started failing because ES reports an unrecognized GROK pattern as a 500, instead of a 400. Not ideal for us, but I'd rather propagate this specific bug than assuming every 500 is a 400 (cc @tonyghiani)

Reply:
Sounds good, not sure why ES returns a 500 when the user mistakenly pass an unacceptable processor config (like invalid pattern), that sounds more a 4xx status code, but I'm ok with forwarding the ES error as far as we still return the error reason to the client

Reply:
++, I've opened a bug for the data management folks after talking to them briefly https://github.com/elastic/elasticsearch/issues/120731